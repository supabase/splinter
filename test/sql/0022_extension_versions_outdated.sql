begin;

  -- 0 issues initially (all extensions should be up to date)
  select * from lint."0022_extension_versions_outdated";

  create extension amcheck version '1.0';

  -- Verify the query structure and column names
  select
    count(*) as total_outdated_extensions
  from lint."0022_extension_versions_outdated";

  -- Test that the query returns proper column structure
  -- This will help ensure the lint is properly formed
  select
    name,
    title,
    level,
    facing,
    categories,
    description,
    detail,
    remediation,
    metadata,
    cache_key
  from lint."0022_extension_versions_outdated";

  drop extension amcheck;

  -- Versions that aren't in pg_available_extension_versions are ignored
  create extension amcheck;
  update pg_extension set extversion = 'foo' where extname = 'amcheck';

  select
    name,
    title,
    level,
    facing,
    categories,
    description,
    detail,
    remediation,
    metadata,
    cache_key
  from lint."0022_extension_versions_outdated";

  drop extension amcheck;
rollback;
