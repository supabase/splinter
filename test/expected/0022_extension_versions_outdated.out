begin;
  -- 0 issues initially (all extensions should be up to date)
  select * from lint."0022_extension_versions_outdated";
 name | title | level | facing | categories | description | detail | remediation | metadata | cache_key 
------+-------+-------+--------+------------+-------------+--------+-------------+----------+-----------
(0 rows)

  create extension amcheck version '1.0';
  -- Verify the query structure and column names
  select
    count(*) as total_outdated_extensions
  from lint."0022_extension_versions_outdated";
 total_outdated_extensions 
---------------------------
                         1
(1 row)

  -- Test that the query returns proper column structure
  -- This will help ensure the lint is properly formed
  select
    name,
    title,
    level,
    facing,
    categories,
    description,
    detail,
    remediation,
    metadata,
    cache_key
  from lint."0022_extension_versions_outdated";
            name             |            title            | level |  facing  | categories |                               description                                |                                                                              detail                                                                               |                                           remediation                                           |                                      metadata                                       |                cache_key                
-----------------------------+-----------------------------+-------+----------+------------+--------------------------------------------------------------------------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------+-------------------------------------------------------------------------------------------------+-------------------------------------------------------------------------------------+-----------------------------------------
 extension_versions_outdated | Extension Versions Outdated | WARN  | EXTERNAL | {SECURITY} | Detects extensions that are not using the default (recommended) version. | Extension `amcheck` is using version `1.0` but version `1.3` is available. Using outdated extension versions may expose the database to security vulnerabilities. | https://supabase.com/docs/guides/database/database-linter?lint=0022_extension_versions_outdated | {"extension_name": "amcheck", "default_version": "1.3", "installed_version": "1.0"} | extension_versions_outdated_amcheck_1.0
(1 row)

  drop extension amcheck;
  -- Versions that aren't in pg_available_extension_versions are ignored
  create extension amcheck;
  update pg_extension set extversion = 'foo' where extname = 'amcheck';
  select
    name,
    title,
    level,
    facing,
    categories,
    description,
    detail,
    remediation,
    metadata,
    cache_key
  from lint."0022_extension_versions_outdated";
 name | title | level | facing | categories | description | detail | remediation | metadata | cache_key 
------+-------+-------+--------+------------+-------------+--------+-------------+----------+-----------
(0 rows)

  drop extension amcheck;
rollback;
